File: proj_curso/API/achadinhos_online/src/main/java/com/example/achadinhos_online/domain/carrinho/Carrinho.java
package com.example.achadinhos_online.domain.carrinho;

import com.example.achadinhos_online.domain.funcionario.Funcionario;
import com.example.achadinhos_online.domain.produto.Produto;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Table(name = "carrinho")
@Entity(name = "Carrinho")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")

public class Carrinho {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "funcionario_id")
    private Funcionario funcionario;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "produto_id")
    private Produto produto;
}

File: projeto_extra/APP/src/app/interfaces/Carrinho.ts
import { Funcionario } from './Funcionario';
import { Produto } from './Produto';

export interface Carrinho {
  id?: number;
  funcionario: Funcionario;
  produto: Produto;
  quantidade: number;
}

export interface CarrinhoEnvio {
  id?: number;
  funcionarioId: number;
  produtoId: number;
  codEditVenda: boolean;
  quantidade: number;
}
